WordPress/WordPressTest/MediaServiceRemoteRESTTests.swift

```diff
@@ -0,0 +1,144 @@
+            }, failure: nil)
+        mockRemoteApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertNotNil(remoteMedia)
+    }
+
+    func testCreateMediaPath() {
+
+        var progress: NSProgress? = nil
+        let expectedPath = "v1.1/sites/\(siteID)/media/new"
+        let media = mockRemoteMedia()
+        mediaServiceRemote.createMedia(media, progress: &progress, success: nil, failure: nil)
+        XCTAssertTrue(mockRemoteApi.postMethodCalled, "Wrong method")
+        XCTAssertEqual(mockRemoteApi.URLStringPassedIn, expectedPath, "Wrong path")
+    }
+
+    func testCreateMedia() {
+
+        let response = ["media" : [["ID" : 1]]]
+        let media = mockRemoteMedia()
+        var progress: NSProgress? = nil
+        var remoteMedia: RemoteMedia? = nil
+        mediaServiceRemote.createMedia(media, progress: &progress, success: {
+            remoteMedia = $0
+            }, failure: nil)
+        mockRemoteApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(media.mediaID, remoteMedia?.mediaID)
+    }
+
+    func testCreateMediaError() {
+
+        let response = ["error" : ["some error"]]
+        let media = mockRemoteMedia()
+        var progress: NSProgress? = nil
+        var errorDescription = ""
+        mediaServiceRemote.createMedia(media, progress: &progress, success:nil, failure: {
+            errorDescription = $0.localizedDescription
+        })
+        mockRemoteApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(errorDescription, response["error"]![0])
+    }
+
+    func testUpdateMediaPath() {
+
+        let media = mockRemoteMedia()
+        let expectedPath = "v1.1/sites/\(siteID)/media/\(media.mediaID)"
+        mediaServiceRemote.updateMedia(media, success: nil, failure: nil)
+        XCTAssertTrue(mockRemoteApi.postMethodCalled, "Wrong method")
+        XCTAssertEqual(mockRemoteApi.URLStringPassedIn, expectedPath, "Wrong path")
+    }
+
+    func testUpdateMedia() {
+
+        let response = ["ID" : 1]
+        let media = mockRemoteMedia()
+        var remoteMedia: RemoteMedia? = nil
+        mediaServiceRemote.updateMedia(media, success: {
+            remoteMedia = $0
+            }, failure: nil)
+        mockRemoteApi.successBlockPassedIn?(response, NSHTTPURLResponse())
+        XCTAssertEqual(media.mediaID, remoteMedia?.mediaID)
+    }
+
+    func testGetMediaLibraryPath() {
+
+        let expectedPath = "v1.1/sites/\(siteID)/media"
+        mediaServiceRemote.getMediaLibraryWithSuccess(nil, failure: nil)
+        XCTAssertTrue(mockRemoteApi.getMethodCalled, "Wrong method")
+        XCTAssertEqual(mockRemoteApi.URLStringPassedIn, expectedPath, "Wrong path")
+    }
+
+    func testGetMediaLibrary() {
+
+        let response = ["media" : [["ID" : 1]]]
+        var remoteMedias = [RemoteMedia]()
+        mediaServiceRemote.getMediaLibraryWithSuccess({
+            if let medias = $0 as? [RemoteMedia] {
```

Comment:
No need to assign to `remoteMedias` since you're only testing if its empty. Simpler to just assign to a bool if the block is called correctly.

If we're going to test that the number of media provided is as expected, that's another thing.